apps/passport-server/src/database/queries/rateLimit.ts

```diff
@@ -0,0 +1,20 @@
+import { Pool } from "postgres-pool";
+import { RateLimit } from "../models";
+import { sqlQuery } from "../sqlQuery";
+
+export async function checkRateLimit(
+  client: Pool,
+  eventType: RateLimit["type"],
+  eventId: string
+): Promise<boolean> {
+  return (
+    await sqlQuery(client, `SELECT take_token($1, $2)`, [eventType, eventId])
+  ).rows[0].take_token;
+}
+
+export async function clearExpiredRateLimits(client: Pool): Promise<void> {
```

Comment:
This is never called outside of tests.  Do you think we need to worry about cleanup here?  Or we can just wait to cleanup if it becomes a problem later.  As I understand it, the cost is 2 DB rows per user.

Reply:
I left it here to prompt a discussion about where the best place to call it from would be. `UserService` could do do it, but that feels wrong. Or it could be called from `startApplication`, but that also feels too high-level.

Reply:
thoughts: It could make sense to create a new `RateLimitService` through which the rest of the services would call `checkRateLimit`. Maybe once every 100 rate limit checks it could call `clearExpiredRateLimits`. I am also not opposed to creating some sort of `onStartup` file and function that does all sorts of generically useful things one would do on startup that don't neatly fit in other places. Though if we do that, it could be hard to track down such things later on.